xray/core/dataset.py

```diff
@@ -1028,6 +1029,58 @@ def sel(self, method=None, **indexers):
         return self.isel(**indexing.remap_label_indexers(self, indexers,
                                                          method=method))
 
+    def isel_points(self, dim='points', **indexers):
+        """Returns a new dataset with each array indexed pointwise along the
+        specified dimension(s).
+
+        This method selects pointwise values from each array and is akin to
+        the NumPy indexing behavior of `arr[[0, 1], [0, 1]]`, except this
+        method does not require knowing the order of each array's dimensions.
+
+        Parameters
+        ----------
+        dim : str, optional
```

Comment:
A neat trick about using `concat` for the underlying implementation here is that it means we can also support supplying all the valid arguments that work for the `dim` argument to concat, e.g., including existing DataArray and pandas.Index objects: http://xray.readthedocs.org/en/stable/generated/xray.concat.html#xray.concat

Might be worth mentioning that.